lib/db.js

```diff
@@ -1178,22 +1178,28 @@ var _executeAuthCreateUserCommand = function(self, username, password, options,
     roles = ['dbOwner'];
   }
 
+  const digestPassword = self.s.topology.lastIsMaster().maxWireVersion >= 7;
+
   // Build the command to execute
   var command = {
     createUser: username,
     customData: customData,
     roles: roles,
-    digestPassword: false
+    digestPassword
   };
 
   // Apply write concern to command
   command = applyWriteConcern(command, { db: self }, options);
 
-  // Use node md5 generator
-  var md5 = crypto.createHash('md5');
-  // Generate keys used for authentication
-  md5.update(username + ':mongo:' + password);
-  var userPassword = md5.digest('hex');
+  let userPassword = password;
+
+  if (!digestPassword) {
+    // Use node md5 generator
```

Comment:
We have open tickets for dropping MD5 to support FIPS environments. Is this md5 segment required by spec, or just something special we cooked up?

Reply:
I believe it is required by SCRAM-SHA-1. Are we getting rid of it in SCRAM-SHA-1?

Reply:
never mind, confirmed with Bernie that this is kosher